File: LLM_Translation/C_programs/insert.c
#include <stdio.h>
#include <stdlib.h>

struct node {
	int key;
	struct node *left, *right;
};

// A utility function to create a new BST node
struct node* newNode(int item)
{
	struct node* temp
		= (struct node*)malloc(sizeof(struct node));
	temp->key = item;
	temp->left = NULL;
	temp->right = NULL;
	return temp;
}

struct node* insert(struct node* node, int key)
{
	/* If the tree is empty, return a new node */
	if (node == NULL)
		return newNode(key);

	/* Otherwise, recur down the tree */
	if (key < node->key)
		node->left = insert(node->left, key);
	else
		node->right = insert(node->right, key);

	/* return the (unchanged) node pointer */
	return node;
}

void main(){
	
}

File: LLM_Translation/Rust_programs_compiled/insert.rs
use std::boxed::Box;

struct Node {
    key: i32,
    left: Option<Box<Node>>,
    right: Option<Box<Node>>,
}

// A utility function to create a new BST node
fn new_node(item: i32) -> Box<Node> {
    Box::new(Node {
        key: item,
        left: None,
        right: None,
    })
}

fn insert(node: Option<Box<Node>>, key: i32) -> Option<Box<Node>> {
    match node {
        None => Some(new_node(key)),
        Some(mut current_node) => {
            if key < current_node.key {
                current_node.left = insert(current_node.left.take(), key);
            } else {
                current_node.right = insert(current_node.right.take(), key);
            }
            Some(current_node)
        }
    }
}

fn main() {}
